martha/martha_v3.js

```diff
@@ -468,22 +460,45 @@ async function marthaV3Handler(req, res) {
         `with auth required '${sendAuth}' and bond provider '${bondProvider}'`
     );
 
+    Object.assign(opt, {
+        requestedFields,
+        httpsMetadataUrl,
+        auth,
+        drsType,
+        sendAuth,
+        bondProvider,
+        accessMethodType,
+        bondSAKeyUrl,
+        bondAccessTokenUrl,
+    });
+}
+
+async function retrieveData(opt) {
```

Comment:
more non-options in this `opt`

Reply:
Unrelated to your changes but it might be helpful to have a link to Michael's Lucid Chart sequence diagram(s) here.